### PR TITLE
user/maven: new package

### PR DESCRIPTION
**Pull request by Natthias**
Couldn't get the update string to match other versions one directory down (for when a new major version is release). Everything else should be fine

### CODE REVIEW

File: user/maven/template.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        "test",
+        "-Drat.skip=true",
+        "-Dmaven.repo.local=/cbuild_cache/repository",
+    )
+
+
+def install(self):
+    self.install_files("apache-maven/out", "usr/lib/", name="maven")
+
+    self.mkdir(self.destdir / "usr/bin")
```
Comment (nekopsykose) [inline user/maven/template.py:67]: install_dir

File: user/maven/template.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "8a24c448d4ac397e6b0c019a4d7250068c02d1cdb553299e6bb71c3ccca78b2c",
+    "7a9cdf674fc1703d6382f5f330b3d110ea1b512b51f1652846d9e4e8a588d766",
+]
+
+
+def prepare(self):
+    self.do(
+        "./bootstrap/bin/mvn",
+        "org.apache.maven.plugins:maven-dependency-plugin:2.8:go-offline",
+        "-Dmaven.repo.local=/cbuild_cache/repository",
```
Comment (q66) [inline user/maven/template.py:29]: "repository" is not a good name for the path

File: user/maven/template.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+pkgname = "maven"
+pkgver = "3.9.9"
+pkgrel = 0
+hostmakedepends = ["openjdk21"]
+# FIXME: depends on unversioned providers are somewhat broken in apk
+# depends = ["virtual:java-jre!openjdk21-jre"]
```
Comment (q66) [inline user/maven/template.py:6]: they should work fine now, as long as one of the providers has a correctly set priority

File: user/maven/update.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+url = "https://dlcdn.apache.org/maven/maven-3"
+pattern = r"(\d\.\d\.\d)"
```
Comment (q66) [inline user/maven/update.py:2]: please use a more concrete pattern, something like `r>([0-9.]+)/<` should do

**Pull request merged**